src/stm32f1/adc.cpp

```diff
@@ -241,58 +254,90 @@ void setup(adc::pins const& p_pin)
     pin = hal::value(p_pin) - 10;
   }
 
-  // Set specified pin to analog input mode
+  // Set specified pin to analog input mode.
   configure_pin({ .port = port, .pin = pin }, input_analog);
+}
+}  // namespace
 
-  // Turns on and calibrates ADC only if its first time power-up
-  if (bit_extract<adc_control_register_2::ad_converter_on>(
-        adc_reg->control_2) == 0) {
-    // Power on adc
-    hal::bit_modify(adc_reg->control_2)
-      .set<adc_control_register_2::ad_converter_on>();
-
-    // Start adc calibration
-    hal::bit_modify(adc_reg->control_2)
-      .set<adc_control_register_2::ad_calibration>();
-
-    // Wait for calibration to complete
-    while (bit_extract<adc_control_register_2::ad_calibration>(
-             adc_reg->control_2) == 1) {
-    }
+adc_peripheral_manager::adc_peripheral_manager(adc_selection p_adc_selection,
+                                               hal::basic_lock& p_lock)
+  : m_lock(&p_lock)
```

Comment:
Default initialize the void* to `nullptr` then reassign it later. Its best that we set these variables to known values before we might potentially use them.